classes/Product.php

```diff
@@ -5814,17 +5815,46 @@ public static function getProductProperties($id_lang, $row, Context $context = n
 
         $combination = new Combination($id_product_attribute);
 
-        if (0 != $combination->unit_price_impact && 0 != $row['unit_price_ratio']) {
-            $unitPrice = ($row['price_tax_exc'] / $row['unit_price_ratio']) + $combination->unit_price_impact;
-            $row['unit_price_ratio'] = $row['price_tax_exc'] / $unitPrice;
+        $baseUnitPrice = 0.0;
+        if (isset($row['unit_price'])) {
+            // Unit price is supposed to be in DB and accessible in the row
+            $baseUnitPrice = (float) $row['unit_price'];
         }
 
-        if (isset($row['unit_price_ratio'])) {
-            $row['unit_price'] = ($row['unit_price_ratio'] != 0 ? $row['price'] / $row['unit_price_ratio'] : 0);
-        } else {
-            $row['unit_price'] = 0.0;
+        // Then if combination has an impact we apply it on unit price
+        if (0 != $combination->unit_price_impact && 0 != $baseUnitPrice) {
+            $baseUnitPrice = $baseUnitPrice + $combination->unit_price_impact;
         }
 
+        // Finally, we apply the currency rate
+        $currencyId = Validate::isLoadedObject($context->currency) ? (int) $context->currency->id : (int) Configuration::get('PS_CURRENCY_DEFAULT');
+        $baseUnitPrice = Tools::convertPrice($baseUnitPrice, $currencyId);
+
+        // Compute price ratio based on initial product price and initial unit price (without taxes, group discount, cart rules)
+        $noSpecificPrice = null;
+        $baseProductPrice = Product::getPriceStatic(
+            (int) $row['id_product'],
+            false,
+            $id_product_attribute,
+            6,
+            null,
+            false,
+            false,
+            $quantity,
+            false,
+            null,
+            null,
+            null,
+            $noSpecificPrice,
+            true,
+            false
+        );
+        $row['unit_price_ratio'] = $baseUnitPrice != 0 ? $baseProductPrice / $baseUnitPrice : 0.0;
```

Comment:
if baseUnitPrice is equal to 0, the baseProductPrice get is useless

Reply:
Thanks